### PR TITLE
Added Bedtools genomecov recipe and removed code from mip.pl

### PR DESCRIPTION
**Pull request by northwestwitch**
The scripts generated by the dry run look ok to me.

### CODE REVIEW

File: lib/MIP/Recipes/Bedtools_genomecov.pm [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+## analysis_bedtools_genomecov
+
+## Function : Calculates coverage on BAM files.
+## Returns  : ""
+## Arguments: $parameter_href, $active_parameter_href, $sample_info_href, $file_info_href, $infile_lane_prefix_href, $job_id_href, $sample_id, $insample_directory, $outsample_directory, $program_name, family_id, $temp_directory, $outaligner_dir
+##          : $parameter_href             => Parameter hash {REF}
+##          : $active_parameter_href      => Active parameters for this analysis hash {REF}
+##          : $sample_info_href           => Info on samples and family hash {REF}
+##          : $file_info_href             => The file_info hash {REF}
+##          : $infile_lane_prefix_href    => Infile(s) without the ".ending" {REF}
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Bedtools_genomecov.pm]: $infile_lane_prefix_href    => -> $infile_lane_prefix_href =>

File: lib/MIP/Recipes/Bedtools_genomecov.pm [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+    ## Add merged infile name after merging all BAM files per sample_id
+    my $infile = $file_info_href->{$sample_id}{merge_infile};
+
+    ## Assign file_tags
+    my $infile_tag =
+      $file_info_href->{$sample_id}{pgatk_baserecalibration}{file_tag};
+    my $outfile_tag =
+      $file_info_href->{$sample_id}{$mip_program_name}{file_tag};
+
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Bedtools_genomecov.pm]: "## Files"

File: lib/MIP/Recipes/Bedtools_genomecov.pm [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    my $infile_prefix  = $infile . $infile_tag;
+    my $outfile_prefix = $infile . $outfile_tag;
+
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Bedtools_genomecov.pm]: Add comment ## Path

File: lib/MIP/Recipes/Bedtools_genomecov.pm [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+              catfile( lc($outaligner_dir), q{coveragereport} ),
+            core_number    => $core_number,
+            process_time   => $time,
+            temp_directory => $temp_directory,
+        }
+    );
+
+    # q{.bam} -> ".b*" for getting index as well)
+    my $infile_path = catfile( $insample_directory,
+        $infile_prefix . substr( $infile_suffix, 0, 2 ) . q{*} );
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Bedtools_genomecov.pm]: q{*} -> $ASTERIX

File: mip.pl [hunk 5]
```diff
@@ -1715,12 +1715,24 @@ BEGIN
...
-if ( $active_parameter{pbedtools_genomecov} > 0 ) {    #Run bedtools genomecov
+# Run bedtools genomecov
+if ( $active_parameter{pbedtools_genomecov} > 0 ) {
 
-    $log->info("[Bedtools genomecov]\n");
+    $log->info( q{[Bedtools genomecov]} . $NEWLINE );
+
+    use MIP::Recipes::Bedtools_genomecov qw{ analysis_bedtools_genomecov };
+
+    my $program_name = lc q{bedtools_genomecov};
```
Comment (henrikstranneheim) [inline mip.pl]: loop label

File: mip.pl [hunk 6]
```diff
@@ -1715,12 +1715,24 @@ BEGIN
...
     foreach my $sample_id ( @{ $active_parameter{sample_ids} } ) {
 
+        ## Assign directories
+        my $insample_directory = catdir( $active_parameter{outdata_dir},
+            $sample_id, $active_parameter{outaligner_dir} );
+        my $outsample_directory = catdir(
+            $active_parameter{outdata_dir},    $sample_id,
+            $active_parameter{outaligner_dir}, q{coveragereport} );
+
         mbedtools_genomecov(
```
Comment (henrikstranneheim) [inline mip.pl]: Are you not calling the wrong sub here? analysis_bedtools_genomecov??

Comment (northwestwitch) [reply mip.pl]: Yes I am, how come I didn't get any error?

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**